File: Niveau 4/3 – Structures de données et Balayages/09 - Temps de travail.py
def main():
  nbIntervales = int(input())
  arr = []
  
  for i in range(nbIntervales):
     a, b = map(int, input().split())
     arr.append((a, b))
     
  arr.sort()
  
  ans = 0
  last = -1
  
  for element in arr:
     start = element[0]
     end = element[1]
     
     if start < last:
        start = last
     
     if start > end:
        continue
        
     last = end
     
     ans += end - start
  
  print(ans)

main()

File: Niveau 4/3 – Structures de données et Balayages/09 - Temps de travail.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

int main()
{
    int nbIntervalles;
    cin >> nbIntervalles;

    vector<pair<int, int>> intervals;

    for (int i = 0; i < nbIntervalles; i++)
    {
        int start, end;
        cin >> start >> end;

        intervals.push_back({start, end});
    }

    sort(intervals.begin(), intervals.end());

    int ans = 0;
    int last = -1;

    for (pair<int, int> &p : intervals)
    {
        int start = p.first;
        int end = p.second;

        if (start < last)
        {
            start = last;
        }

        if (start > end)
        {
            continue;
        }

        last = end;
        ans += end - start;
    }

    cout << ans;
}